lib/symbol-provider.coffee

```diff
@@ -206,11 +209,21 @@ class SymbolProvider
   fuzzyFilter: (symbolList, buffer, {bufferPosition, prefix}) ->
     # Probably inefficient to do a linear search
     candidates = []
+
+    if @useAlternateScoring
+      fuzzaldrinProvider = fuzzaldrinPlus
+      # This allows to pre-compute and re-use some quantities derived from prefix such as
+      # Uppercase, lowercase and a version of prefix without optional characters.
+      prefixCache = fuzzaldrinPlus.prepQuery(prefix)
+    else
+     fuzzaldrinProvider = fuzzaldrin
+     prefixCache = null
+
     for symbol in symbolList
       text = (symbol.snippet or symbol.text)
       continue unless text and prefix[0].toLowerCase() is text[0].toLowerCase() # must match the first char!
-      score = fuzzaldrin.score(text, prefix)
-      score *= @getLocalityScore(bufferPosition, symbol.bufferRowsForBuffer?(buffer))
+      score = fuzzaldrinProvider.score(text, prefix, prefixCache)
+      if @useBufferProximity then score *= @getLocalityScore(bufferPosition, symbol.bufferRowsForBuffer?(buffer))
       candidates.push({symbol, score, locality, rowDifference}) if score > 0
```

Comment:
@benogle I'm pretty sure `locality` and `rowDifference` are remains from before `getLocalityScore` was it's own function.